locallib.php

```diff
@@ -758,3 +758,184 @@ function local_ltiprovider_add_user_to_group($tool, $user) {
+                                    $userobj = $DB->get_record('user', array('id' => $userobj->id));
+
+                                    $userphotos[$userobj->id] = $member->user_image;
+                                    // Trigger event.
+                                    $event = \core\event\user_created::create(
+                                        array(
+                                            'objectid' => $userobj->id,
+                                            'relateduserid' => $userobj->id,
+                                            'context' => context_user::instance($userobj->id)
+                                        )
+                                    );
+                                    $event->trigger();
+
+                                    $currentusers[] = $userobj->id;
+                                }
+                            }
+                            // 1 -> Enrol and unenrol, 2 -> enrol
+                            if ($tool->syncmode == 1 or $tool->syncmode == 2) {
+                                // Enroll the user in the course. We don't know if it was previously unenrolled.
+                                $roles = explode(',', strtolower($member->roles));
+                                local_ltiprovider_enrol_user($tool, $userobj, $roles, true);
+                            }
+                        }
+                        // Now we check if we have to unenrol users for keep both systems sync.
+                        if ($tool->syncmode == 1 or $tool->syncmode == 3) {
+                            // Unenrol users also.
+                            $context = context_course::instance($tool->courseid);
+                            $eusers = get_enrolled_users($context);
+                            foreach ($eusers as $euser) {
+                                if (!in_array($euser->id, $currentusers)) {
+                                    local_ltiprovider_unenrol_user($tool, $euser);
+                                }
+                            }
+                        }
+                    } else {
+                        mtrace('Error recived from the remote system: ' . $data->statusinfo->codemajor . ' ' . $data->statusinfo->severity . ' ' . $data->statusinfo->codeminor);
+                    }
+                } else {
+                    mtrace('Error parsing the XML received' . substr($response, 0, 125) . '... (Displaying only 125 chars)');
+                }
+            } else {
+                mtrace('No response received from ' . $user->membershipsurl);
+            }
+        }
+        set_config('membershipslastsync-' . $tool->id, $timenow, 'local_ltiprovider');
+        mtrace('Finished sync of member using the memberships service');
+    }
+    return array('userphotos'=>$userphotos, 'consumers'=>$consumers, 'response'=>$response);
+}
+
+function local_ltiprovider_membership_service_update_userphotos($userphotos) {
```

Comment:
phpdocs for this function here too please

Reply:
Done!